### DISCUSSION TITLE
multiple issues with TuYa smart knob ERS-10TZBVK-AA

### DISCUSSION THREAD

**Discussion by mkaut** [Koenkk/zigbee2mqtt#19136] [Devices]
I have the knob paired and reporting, but I have several issues with it.

First, the [device information page](https://www.zigbee2mqtt.io/devices/ERS-10TZBVK-AA.html) recommends adding it to a group. I therefore made a group, but adding the device to it (using the web frontend) always fails.

I also wanted to bind it to my Namron 3802966 bulb, but that fails as well (I have selected LevelCtrl and onOff clusters and get an error message for both of them). I tried to do it in both operation modes ("event" and "command").
By the way, I had the knob bound to the Coordinator's endpoint 1 for clusters OnOff (or was it LevelCtrl?) and PowerCfg. Should it be like this? (I assume I added this myself when I bought the knob some time ago, but I do not recall it and therefore do not know if it is supposed to be like this. I unbound it during my testing and so far did not rebound it...)

As a side node, I struggled switching it from "event" to "command" operation mode - it failed several times, but worked at the end

Could the source of the problems be that this is a battery device that does not readily receive commands?

**Comment by deviantintegral** [comment]
I'd also had trouble with group binding, until I had an idea based on reading https://github.com/Koenkk/zigbee2mqtt/issues/12397. I tried adding the button to the group itself, and then making sure it was in command mode, and then it worked!

<img width="1141" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/255023/083467a0-a4db-43c2-94b0-4c2bb6d2c61c">

On / off and brightness work super smoothly. I haven't tested it with a group with colour bulbs yet for the press-and-hold hue_move command.

_however_, the state of the light group isn't updated correctly. I can turn them off, and the individual bulbs turn off, but the group still shows up as on. Unfortunately for me, that means that Adaptive Lighting kicks in eventually and turns them back on.

This looks to be because the dial has a permanent state of `ON`:

```json
{
  "action": "",
  "action_rate": null,
  "action_step_size": null,
  "action_transition_time": null,
  "battery": 92,
  "last_seen": "2023-12-18T19:39:34.026Z",
  "linkquality": 76,
  "operation_mode": "command",
  "state": "ON",
  "voltage": 2900
}
```

This then causes `Groups.areAllMembersOff()` to always return TRUE.

It's not clear to me where the state attribute is getting set.

**Reply by deviantintegral** [reply]
I figured it out! The problem was an extra `state` attribute in `state.json`:

```json
    "0x70ac08fffe266333": {
        "operation_mode": "command",
        "battery": 92,
        "voltage": 2900,
        "last_seen": "2023-12-19T16:02:39.400Z",
        "linkquality": 72,
        "state": "ON"
    },
```

I shut down zigbee2mqtt, removed `"state": "ON"` (don't forget the comma on the line before!), and restarted. Then my group behaved correctly.

I wasn't able to reproduce this in a separate z2m install, so I think this extra attribute may have been a bug that's since been fixed. The most confusing part is that re-pairing didn't fix this, so I suppose I would have had to fully remove the device to clear the bad entry.

**Reply by deviantintegral** [reply]
I was only partially correct. The way to trigger this was to send an on or off command to the group itself. Then, z2m was setting the state on the individual devices. I thought the right fix would be `optimistic: false`, but in practice it was setting `off_state: 'last_member_state'`.

**Reply by williamalban** [reply]
Can you please explain better how did you get it into the group. I am having the same issue, error when I try to add the KNOB to any groups, and also, it won't unbind from the coordinator
<img width="1167" alt="Captura de Tela 2024-07-18 às 19 00 01" src="https://github.com/user-attachments/assets/3b633955-34a5-4b24-bd4a-96a5b7fe4a1e">

**Reply by williamalban** [reply]
I finally was able to add the knob into the group, what I did was click "add to group" than do the triple click on the knob so it switches from evento mode to command mode. But I still can't unbind from the coordinator.
@deviantintegral Please, what you did to get it to work specially to get brightness to work super smoothly????

**Reply by SashaBusinaro** [reply]
I am also interested in understanding how to make the brightness change smooth.

**Reply by deviantintegral** [reply]
Here’s what I have in the group:

<img width="1373" alt="image" src="https://github.com/user-attachments/assets/97f652e1-b95e-4936-9745-e56238b503bd">

And in the binding table:

<img width="1373" alt="image" src="https://github.com/user-attachments/assets/59d1d26f-51bb-4390-991e-ea2b539726ec">

I didn’t fully unbind from the coordinator - in fact, I would check to see if it works just by setting up the group as in the first screenshot.

**Reply by f22raptorroland** [reply]
**Oh wow, it does work just by setting it up like in the first screenshot.**

Thank you for all the info guys!

SO
1.  I added the lights to a group
2. Click add to group on knob then quickly triple-click (it adds it to the group, and a binding automatically appeared)
3. triple-click again (to change back the mode) and it works!

binding:
![image](https://github.com/user-attachments/assets/5c6fadec-edb5-4e82-b65d-c6bd6c1dd757)
